jax/_src/checkify.py

```diff
@@ -1264,9 +1062,177 @@ def checkify(fun: Callable[..., Out],
   """
   @traceback_util.api_boundary
   def checked_fun(*args, **kwargs):
-    args_flat, in_tree = tree_flatten((args, kwargs))
-    f, out_tree = flatten_fun(lu.wrap_init(fun), in_tree)
-    error, out_flat = checkify_flat(f, errors, *args_flat)
-    out = tree_unflatten(out_tree(), out_flat)
-    return error, out
+    # stage:
+    fun = lu.wrap_init(f, kwargs)
```

Comment:
I think we want to use `flatten_fun` here, not `flatten_fun_nokwargs` so we trace the kwargs (unless they opt out by closing over or using partial). We may want to eventually support static_argnums/names. This is consistent with other transformations.